modules/core/src/system.cpp

```diff
@@ -1834,6 +1834,8 @@ static void WINAPI opencv_fls_destructor(void* pData)
 #endif // CV_USE_FLS
 #endif // _WIN32
 
+static TlsAbstraction* g_force_initialization_of_TlsAbstraction = getTlsAbstraction();
```

Comment:
Why do this rather than fix the startup threading of getTlsAbstraction() If someone is going to code a fix for this issue, risk code change, and the testing...I suggest a better use of effort is to fix the actual start issue rather than a partial workaround for the startup issue.

And this code change only assists with making that function call before main(). It doesn't help with other static calls, lambdas, Windows DLLs and DllMain(), probably Linux shared libraries, etc.

Reply:
Which fixes do you want to [add](https://github.com/opencv/opencv/blob/3.4.15/modules/core/src/system.cpp#L1479-L1493)?

Reply:
I would not add this code into the opencv project. The specific user that reported a specific problem in their specific project due to a specific race-condition should use this kind of code in their project. Not OpenCV itself.
https://en.cppreference.com/w/cpp/language/siof

This hack (which it seems you have already merged) doesn't work across compile units. This hack create yet another race condition. The race condition that existed before continues to exist. And now this hack creates yet another.

A reliable "fix" requires rethinking the TLS startup (and shutdown) behavior. The startup can likely be fixed in C++11. The shutdown is very difficult (maybe impossible) to fix without a "shutdown manager" which requires classes/objects requiring shutdown handling to formall register themselves with the shutdown manager.